src/directions/events.ts

```diff
@@ -8,44 +8,45 @@ export class MapLibreGlDirectionsEvented {
 
   protected readonly map: Map;
 
-  private listeners: Partial<Record<keyof MapLibreGlDirectionsEventType, ((ev: never) => void)[]>> = {};
-  private oneTimeListeners: Partial<Record<keyof MapLibreGlDirectionsEventType, ((ev: never) => void)[]>> = {};
+  private listeners: ListenersStore = {};
+  private oneTimeListeners: ListenersStore = {};
 
   protected fire<T extends keyof MapLibreGlDirectionsEventType>(event: MapLibreGlDirectionsEventType[T]) {
     event.target = this.map;
 
-    this.listeners[event.type]?.forEach((listener) => listener(event));
-    this.oneTimeListeners[event.type]?.forEach((listener) => {
+    const type: T = event.type as T;
+
+    this.listeners[type]?.forEach((listener) => listener(event));
+    this.oneTimeListeners[type]?.forEach((listener) => {
       listener(event);
 
-      const index = this.oneTimeListeners[event.type]?.indexOf(listener);
-      if (index && ~index) this.oneTimeListeners[event.type]?.splice(index, 1);
+      const index = this.oneTimeListeners[type]?.indexOf(listener);
+      if (index !== undefined && ~index) this.oneTimeListeners[type]?.splice(index, 1);
     });
   }
 
   /**
    * Registers an event listener.
    */
-  on<T extends keyof MapLibreGlDirectionsEventType>(type: T, listener: (ev: MapLibreGlDirectionsEventType[T]) => void) {
-    (this.listeners[type] = this.listeners[type] ?? ([] as ((ev: never) => void)[])).push(listener);
+  on<T extends keyof MapLibreGlDirectionsEventType>(type: T, listener: MapLibreGlDirectionsEventListener<T>) {
+    this.listeners[type] = this.listeners[type] ?? [];
+    this.listeners[type]!.push(listener);
```

Comment:
Just for readability or what's the point of the change?

Reply:
Because TS isn't smart enough to process in-place assignment and reduces output type to never[], so the splitting assignment is the least hacky workaround to make TS happy without inserting something like `as any`